lib/plugins/plugin/plugin.js

```diff
@@ -0,0 +1,415 @@
+              name: {
+                usage: 'The plugin name',
+                required: true,
+                shortcut: 'n',
+              },
+              version: {
+                usage: 'The plugin version',
+                shortcut: 'v',
+              },
+            },
+          },
+        },
+      },
+      uninstall: {
+        usage: 'Uninstall and remove a plugin from your service',
+        commands: {
+          plugin: {
+            usage: 'Uninstall and remove a plugin from your service',
+            lifecycleEvents: [
+              'plugin',
+            ],
+            options: {
+              name: {
+                usage: 'The plugin name',
+                required: true,
+                shortcut: 'n',
+              },
+            },
+          },
+        },
+      },
+    };
+
+    this.hooks = {
+      'plugin:install:install': () => BbPromise.bind(this)
+        .then(this.install)
+        .then(this.trackPluginInstall),
+      'install:plugin:plugin': () => BbPromise.bind(this)
+        .then(this.install)
+        .then(this.trackPluginInstall),
+      'plugin:uninstall:uninstall': () => BbPromise.bind(this)
+        .then(this.uninstall)
+        .then(this.trackPluginUninstall),
+      'uninstall:plugin:plugin': () => BbPromise.bind(this)
+        .then(this.uninstall)
+        .then(this.trackPluginUninstall),
+      'plugin:list:list': () => BbPromise.bind(this)
+        .then(this.list)
+        .then(this.trackPluginList),
+      'plugin:search:search': () => BbPromise.bind(this)
+        .then(this.search)
+        .then(this.trackPluginSearch),
+    };
+  }
+
+  install() {
+    return BbPromise.bind(this)
+      .then(this.validate)
+      .then(this.getPlugins)
+      .then((plugins) => {
+        const pluginName = this.options.name;
+        const plugin = plugins.find((item) => item.name === pluginName);
+
+        if (plugin) {
+          const servicePath = this.serverless.config.servicePath;
+          const packageJsonFilePath = path.join(servicePath, 'package.json');
+
+          // check if package.json is already present. Otherwise create one
+          if (!fs.existsSync(packageJsonFilePath)) {
+            this.serverless.cli
+              .log('Creating an empty package.json file in your service directory');
+
+            const packageJsonFileContent = {
+              name: this.serverless.service.service,
+              description: '',
+              version: '0.1.0',
+              dependencies: {},
+              devDependencies: {},
+            };
+
+            fse.writeJsonSync(packageJsonFilePath, packageJsonFileContent);
+          }
+
+          // decide the plugin version
+          const pluginVersion = this.options.version || 'latest';
+
+          // install the package through npm
+          this.serverless.cli.log(
+            `Installing plugin "${pluginName}@${pluginVersion}" (this might take a few seconds...)`
+          );
+
+          childProcess
```

Comment:
@horike37 please avoid doing these kinds of things synchronously. If you need to wait on something to complete it is better to use a Promise and make it part of the async chain.